### ISSUE TITLE
Rewriting browser-phone in typescript and react js for easy ui modification and integration

### ISSUE THREAD

**Issue by KellyKinyama** [InnovateAsterisk/Browser-Phone#574]
I tried to rewrite browser-phone in typescript but the task is daunting and not trivial. This is a great project. But for integration and ui modification in code is not easy

**Comment by InnovateAsterisk** [comment]
Yes! it's a significant project to get it over to TypeScript... good thing we are doing that for you. 🎉 We should have the first (converted) release at siperb.com early next year.

**Comment by KellyKinyama** [comment]
We want to buy some cups coffee on this project

**Comment by InnovateAsterisk** [comment]
While I do love a coffee 😀. What i'm most interested in is the success of Siperb. (https://www.siperb.com/) It's a clone of Browser Phone, but made to be provided as a service. Right now we are looking for people who are willing to try it out, and provide feedback, also let us know if it's the kind of thing that would work for you, if not what's missing etc.

Parts of it will be open source - like the core scripts and the web UI (React), but the mobile apps and all that goes on in the cloud will not be.

The idea with Siperb is that it's all provided for you, free (and for you to view the open source). It should be easier for you to just use the Siperb Apps and Proxy service, than having to get your own instances together, learn how proxies work, and connect A with B... It's all done and it costs nothing.  

Right now the main advantage here is, if we host it all, we can then push calls to you.

Pretty soon, we will have:
- Transcoding (RTP to DTLS-SRTP)
- Direct Websocket to own Asterisk Server connections
- Event Hooks, and customizations  (like window UI, alerts etc)

Down the line we have some radicle AI features we want to fold in, like:
- Voicemail messages will be transcribed, and pushed to your mobile app... hmm nice but boring right?
- How about AI transcribing those call recordings for you, maybe it can just give you a summary of the call in text, as a message pushed to your phone... cool hey
- Wrong language??... no problem just flip the translation switch, and we will do on-the-fly translation for you... Now that's cool.
- Even cooler, let's say you are working a call center, the AI is listening in on the call, a special window sits in front of you, and suggest things to say, coaching you to say the right thing. Nice!
- Take that even further, just give the AI a voice, and let it take the call, all good!
- Totally blow you mind - train an avatar and next thing the AI agent, not only has a voice, but a face too!

🤯

**Comment by KellyKinyama** [comment]
We are ready to participate as we are looking for a call center solution. I navigated and registered at (https://www.siperb.com/) but it was not clear on how to proceed

**Comment by InnovateAsterisk** [comment]
This is the onboarding category:
https://www.siperb.com/kb/topics/onboarding/

this is also a good article to read:
https://www.siperb.com/kb/article/welcome-to-the-siperb-beta-program/

As for call center, I would suggest two things: 
You can embed the phone in an iframe, and tap into the eventing system. This way you can build your own call dispositions, and click to dial functions, while calling is handled for you. Currently each agent would have to be provisioned separately, but fairly soon we will have the management layer for bigger deployments.
